src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -44,57 +44,54 @@ typedef struct PartitionedResultDestReceiver
 	/* public DestReceiver interface */
 	DestReceiver pub;
 
-	/* partition file $i is stored at file named $resultIdPrefix_$i. */
-	char *resultIdPrefix;
+	/* on lazy startup we only startup the DestReceiver once they receive a tuple */
+	bool lazyStartup;
 
-	/* use binary copy or just text copy format? */
-	bool binaryCopy;
-
-	/* used for deciding which partition a shard belongs to. */
-	CitusTableCacheEntry *shardSearchInfo;
-
-	MemoryContext perTupleContext;
-
-	/* how does stream tuples look like? */
-	TupleDesc tupleDescriptor;
+	/* startup information for use in lazy startup */
+	struct
+	{
+		int operation;
+		TupleDesc tupleDescriptor;
```

Comment:
can we add comments to these? 

(preferably with better grammar than "how does stream tuples look like" 😋 )

Reply:
I wrote a novel about its use